<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get the PAR-Y Regions From a Seqinfo Object — parY • transmogR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get the PAR-Y Regions From a Seqinfo Object — parY"><meta name="description" content="Define the Pseudo-Autosomal Regions from a Seqinfo Object"><meta property="og:description" content="Define the Pseudo-Autosomal Regions from a Seqinfo Object"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="default" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">transmogR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><a class="dropdown-item" href="../articles/creating_a_new_reference.html">Creating a variant modified reference</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/smped/transmogR" aria-label="View on github"><span class="fa fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get the PAR-Y Regions From a Seqinfo Object</h1>
      <small class="dont-index">Source: <a href="https://github.com/smped/transmogR/blob/gh-actions/R/parY.R" class="external-link"><code>R/parY.R</code></a></small>
      <div class="d-none name"><code>parY-methods.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Define the Pseudo-Autosomal Regions from a Seqinfo Object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">parY</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'Seqinfo'</span></span>
<span><span class="fu">parY</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'character'</span></span>
<span><span class="fu">parY</span><span class="op">(</span><span class="va">x</span>, prefix <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>A Seqinfo object or any of named build. If passing
a character vector, <code><a href="https://rdrr.io/r/base/match.arg.html" class="external-link">match.arg()</a></code> will be used to match the build.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Not used</p></dd>


<dt id="arg-prefix">prefix<a class="anchor" aria-label="anchor" href="#arg-prefix"></a></dt>
<dd><p>Optional prefix to place before chromosome names. Can only be
NULL, "" or "chr"</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A GenomicRanges object</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Using a seqinfo object based on either hg38, hg19, CHM13.v2 or their
variations, create a GRanges object with the Pseudo-Autosomal Regions from
the Y chromosome for that build.
The length of the Y chromosome on the seqinfo object is used to determine
the correct genome build when passing a Seqinfo object.
Otherwise</p>
<p>An additional mcols column called PAR will indicate PAR1 and PAR2</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomeInfoDb" class="external-link">GenomeInfoDb</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    seqnames <span class="op">=</span> <span class="st">"chrY"</span>, seqlengths <span class="op">=</span> <span class="fl">59373566</span>, genome <span class="op">=</span> <span class="st">"hg19_only_chrY"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">parY</span><span class="op">(</span><span class="va">sq</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 2 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames            ranges strand |         PAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrY     10001-2649520      * |        PAR1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrY 59034050-59363566      * |        PAR2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from hg19_only_chrY genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## PAR regions for CHM13 are also available</span></span></span>
<span class="r-in"><span><span class="va">sq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    seqnames <span class="op">=</span> <span class="st">"chrY"</span>, seqlengths <span class="op">=</span> <span class="fl">62460029</span>, genome <span class="op">=</span> <span class="st">"CHM13"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">parY</span><span class="op">(</span><span class="va">sq</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 2 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames            ranges strand |         PAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrY         1-2458320      * |        PAR1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrY 62122809-62460029      * |        PAR2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from CHM13 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Or just call by name</span></span></span>
<span class="r-in"><span><span class="fu">parY</span><span class="op">(</span><span class="st">"GRCh38"</span>, prefix <span class="op">=</span> <span class="st">"chr"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 2 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames            ranges strand |         PAR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chrY     10001-2781479      * |        PAR1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chrY 56887903-57217415      * |        PAR2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from GRCh38 genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/smped" class="external-link">Stevie Pederson</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer></div>





  </body></html>

