<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Obtain Splice-Junctions from Exons and Transcripts — sjFromExons • transmogR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Obtain Splice-Junctions from Exons and Transcripts — sjFromExons"><meta property="og:description" content="Using GRanges defining exons and transcripts, find the splice-junctions"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">transmogR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/creating_a_new_reference.html">Creating a variant modified reference</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/smped/transmogR" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Obtain Splice-Junctions from Exons and Transcripts</h1>
    <small class="dont-index">Source: <a href="https://github.com/smped/transmogR/blob/HEAD/R/sjFromExons.R" class="external-link"><code>R/sjFromExons.R</code></a></small>
    <div class="hidden name"><code>sjFromExons.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Using GRanges defining exons and transcripts, find the splice-junctions</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">sjFromExons</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  rank_col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"exon_number"</span>, <span class="st">"exon_rank"</span><span class="op">)</span>,</span>
<span>  tx_col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"transcript_id"</span>, <span class="st">"tx_id"</span><span class="op">)</span>,</span>
<span>  extra_cols <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  don_len <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  acc_len <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  as <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GRanges"</span>, <span class="st">"GInteractions"</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>GRanges object with exons and transcripts. A column indicating the
position (or rank) of each exon within the transcript must be included.</p></dd>


<dt>rank_col</dt>
<dd><p>The column containing the position of each exons within the
transcript</p></dd>


<dt>tx_col</dt>
<dd><p>The column containing unique transcript-level identifiers</p></dd>


<dt>extra_cols</dt>
<dd><p>Can be a vector of column names to return beyond rank_col
and tx_col. By default all columns are returned (extra_cols = "all").</p></dd>


<dt>don_len, acc_len</dt>
<dd><p>Length of donor and acceptor sites respectively</p></dd>


<dt>as</dt>
<dd><p>Return as a set of GenomicRanges, or with each splice junction
annotated as a GenomicInteraction</p></dd>


<dt>...</dt>
<dd><p>Not used</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A GRanges object with requested columns, and an additional column, 'site',
annotating each region as a donor or acceptor site.</p>


<p>Alternatively, by specifying as = "GInteractions", the junctions can be
returned with each splice junction annotated as a GenomicInteraction.
This can make the set of junctions easier to interpret for a given
transcript.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>A canonical splice junction consists of a donor site and an acceptor site at
each end of an intron, with a branching site somewhere wthin the intron.
Canonical donor sites are 8nt long with the the first two bases being exonic
and the next 6 being derived form intronic sequences.
Canonical acceptor sites are 5nt long with the first four bases being
intronic and the final base being the first base of the next exon.</p>
<p>This functions uses each set of exons within a transcript to identify both
donor and acceptor sites. Branch sites are not identified.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gtf_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="st">"transcript_id"</span>, <span class="st">"transcript_name"</span>, <span class="st">"gene_id"</span>, <span class="st">"gene_name"</span>, <span class="st">"exon_number"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/GFFFile-class.html" class="external-link">import.gff</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>   <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/gencode.v44.subset.gtf.gz"</span>, package <span class="op">=</span> <span class="st">"transmogR"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>   feature.type <span class="op">=</span> <span class="st">"exon"</span>, colnames <span class="op">=</span> <span class="va">gtf_cols</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sj</span> <span class="op">&lt;-</span> <span class="fu">sjFromExons</span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sj</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 730 ranges and 6 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames        ranges strand |        site     transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;       &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr1 182745-182752      + |       donor ENST00000624431.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr1 183128-183132      + |    acceptor ENST00000624431.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr1 183215-183222      + |       donor ENST00000624431.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr1 183490-183494      + |    acceptor ENST00000624431.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr1 183570-183577      + |       donor ENST00000624431.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...           ...    ... .         ...               ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [726]     chr1 810061-810068      - |       donor ENST00000590817.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [727]     chr1 810170-810174      - |    acceptor ENST00000447500.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [728]     chr1 817367-817374      - |       donor ENST00000447500.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [729]     chr1 827664-827671      - |       donor ENST00000635509.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [730]     chr1 827664-827671      - |       donor ENST00000634337.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         transcript_name            gene_id       gene_name exon_number</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;character&gt;        &lt;character&gt;     &lt;character&gt;   &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]    DDX11L17-201  ENSG00000279928.2        DDX11L17           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]    DDX11L17-201  ENSG00000279928.2        DDX11L17           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]    DDX11L17-201  ENSG00000279928.2        DDX11L17           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]    DDX11L17-201  ENSG00000279928.2        DDX11L17           3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]    DDX11L17-201  ENSG00000279928.2        DDX11L17           3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...             ...                ...             ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [726] ENST00000590817  ENSG00000230092.8 ENSG00000230092           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [727] ENST00000447500  ENSG00000290784.1 ENSG00000290784           2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [728] ENST00000447500  ENSG00000290784.1 ENSG00000290784           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [729] ENST00000635509 ENSG00000230021.10 ENSG00000230021           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [730] ENST00000634337 ENSG00000230021.10 ENSG00000230021           1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Or to simplify shared splice junctions across multiple transcripts</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/smped/extraChIPs" class="external-link">extraChIPs</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Registered S3 method overwritten by 'ggside':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   method from   </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   +.gg   ggplot2</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/extraChIPs/man/chopMC.html" class="external-link">chopMC</a></span><span class="op">(</span><span class="va">sj</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 196 ranges and 6 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames        ranges strand |        site</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]     chr1 182745-182752      + |       donor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]     chr1 183128-183132      + |    acceptor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]     chr1 183215-183222      + |       donor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]     chr1 183490-183494      + |    acceptor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]     chr1 183570-183577      + |       donor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...      ...           ...    ... .         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [192]     chr1 808623-808627      - |    acceptor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [193]     chr1 810061-810068      - |       donor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [194]     chr1 810170-810174      - |    acceptor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [195]     chr1 817367-817374      - |       donor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [196]     chr1 827664-827671      - |       donor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               transcript_id                 transcript_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                             &lt;CharacterList&gt;                 &lt;CharacterList&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]                   ENST00000624431.2                    DDX11L17-201</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]                   ENST00000624431.2                    DDX11L17-201</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]                   ENST00000624431.2                    DDX11L17-201</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]                   ENST00000624431.2                    DDX11L17-201</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]                   ENST00000624431.2                    DDX11L17-201</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...                                 ...                             ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [192] ENST00000447500.4,ENST00000590817.3 ENST00000447500,ENST00000590817</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [193] ENST00000447500.4,ENST00000590817.3 ENST00000447500,ENST00000590817</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [194]                   ENST00000447500.4                 ENST00000447500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [195]                   ENST00000447500.4                 ENST00000447500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [196] ENST00000635509.2,ENST00000634337.2 ENST00000635509,ENST00000634337</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                       gene_id                       gene_name</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                               &lt;CharacterList&gt;                 &lt;CharacterList&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]                     ENSG00000279928.2                        DDX11L17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]                     ENSG00000279928.2                        DDX11L17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]                     ENSG00000279928.2                        DDX11L17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]                     ENSG00000279928.2                        DDX11L17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]                     ENSG00000279928.2                        DDX11L17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...                                   ...                             ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [192]   ENSG00000290784.1,ENSG00000230092.8 ENSG00000290784,ENSG00000230092</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [193]   ENSG00000290784.1,ENSG00000230092.8 ENSG00000290784,ENSG00000230092</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [194]                     ENSG00000290784.1                 ENSG00000290784</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [195]                     ENSG00000290784.1                 ENSG00000290784</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [196] ENSG00000230021.10,ENSG00000230021.10 ENSG00000230021,ENSG00000230021</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           exon_number</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;IntegerList&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]             1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]             2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]             2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]             3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]             3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...           ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [192]           3,2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [193]           2,1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [194]             2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [195]             1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [196]           1,1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Splice Junctions can also be returned as a GInteractions object with</span></span></span>
<span class="r-in"><span><span class="co">## anchorOne as the donor &amp; anchorTwo as the acceptor sites</span></span></span>
<span class="r-in"><span><span class="fu">sjFromExons</span><span class="op">(</span><span class="va">gtf</span>, as <span class="op">=</span> <span class="st">"GInteractions"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GInteractions object with 365 interactions and 5 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         seqnames1       ranges1     seqnames2       ranges2 |     transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;Rle&gt;     &lt;IRanges&gt;         &lt;Rle&gt;     &lt;IRanges&gt; |       &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]      chr1 182745-182752 ---      chr1 183128-183132 | ENST00000624431.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]      chr1 183215-183222 ---      chr1 183490-183494 | ENST00000624431.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]      chr1 183570-183577 ---      chr1 183736-183740 | ENST00000624431.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]      chr1 183900-183907 ---      chr1 183977-183981 | ENST00000624431.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]      chr1 724359-724366 ---      chr1 724714-724718 | ENST00000440782.3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...       ...           ... ...       ...           ... .               ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [361]      chr1 817367-817374 ---      chr1 774280-774284 | ENST00000447500.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [362]      chr1 827664-827671 ---      chr1 801163-801167 | ENST00000635509.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [363]      chr1 827664-827671 ---      chr1 805891-805895 | ENST00000634337.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [364]      chr1 720026-720033 ---      chr1 805891-805895 | ENST00000414688.6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [365]      chr1 711705-711712 ---      chr1 808623-808627 | ENST00000635509.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         transcript_name            gene_id       gene_name        sj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             &lt;character&gt;        &lt;character&gt;     &lt;character&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [1]    DDX11L17-201  ENSG00000279928.2        DDX11L17         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [2]    DDX11L17-201  ENSG00000279928.2        DDX11L17         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [3]    DDX11L17-201  ENSG00000279928.2        DDX11L17         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [4]    DDX11L17-201  ENSG00000279928.2        DDX11L17         4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     [5]       CICP3-201  ENSG00000229376.3           CICP3         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     ...             ...                ...             ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [361] ENST00000447500  ENSG00000290784.1 ENSG00000290784         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [362] ENST00000635509 ENSG00000230021.10 ENSG00000230021         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [363] ENST00000634337 ENSG00000230021.10 ENSG00000230021         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [364] ENST00000414688 ENSG00000230021.10 ENSG00000230021         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [365] ENST00000635509 ENSG00000230021.10 ENSG00000230021         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   regions: 196 ranges and 0 metadata columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://github.com/smped" class="external-link">Stevie Pederson</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

