<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Mogrify a transcriptome using a set of variants — transmogrify • transmogR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Mogrify a transcriptome using a set of variants — transmogrify"><meta property="og:description" content="Use a set of SNPs, insertions and deletions to modify a reference
transcriptome"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">transmogR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/creating_a_new_reference.html">Creating a variant modified reference</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/smped/transmogR" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Mogrify a transcriptome using a set of variants</h1>
    <small class="dont-index">Source: <a href="https://github.com/smped/transmogR/blob/HEAD/R/transmogrify.R" class="external-link"><code>R/transmogrify.R</code></a></small>
    <div class="hidden name"><code>transmogrify-methods.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Use a set of SNPs, insertions and deletions to modify a reference
transcriptome</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">transmogrify</span><span class="op">(</span><span class="va">x</span>, <span class="va">var</span>, <span class="va">exons</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for XStringSet,GRanges,GRanges</span></span>
<span><span class="fu">transmogrify</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">var</span>,</span>
<span>  <span class="va">exons</span>,</span>
<span>  alt_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>  trans_col <span class="op">=</span> <span class="st">"transcript_id"</span>,</span>
<span>  omit_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tag <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  var_tags <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  var_sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for BSgenome,GRanges,GRanges</span></span>
<span><span class="fu">transmogrify</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">var</span>,</span>
<span>  <span class="va">exons</span>,</span>
<span>  alt_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>  trans_col <span class="op">=</span> <span class="st">"transcript_id"</span>,</span>
<span>  omit_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tag <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  var_tags <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  var_sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for BSgenome,VcfFile,GRanges</span></span>
<span><span class="fu">transmogrify</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">var</span>,</span>
<span>  <span class="va">exons</span>,</span>
<span>  alt_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>  trans_col <span class="op">=</span> <span class="st">"transcript_id"</span>,</span>
<span>  omit_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tag <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  var_tags <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  var_sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">which</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for XStringSet,VcfFile,GRanges</span></span>
<span><span class="fu">transmogrify</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">var</span>,</span>
<span>  <span class="va">exons</span>,</span>
<span>  alt_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>  trans_col <span class="op">=</span> <span class="st">"transcript_id"</span>,</span>
<span>  omit_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tag <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  var_tags <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  var_sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">which</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>Reference genome as either a DNAStringSet or BSgenome</p></dd>


<dt>var</dt>
<dd><p>GRanges object containing the variants</p></dd>


<dt>exons</dt>
<dd><p>GRanges object with ranges representing exons</p></dd>


<dt>...</dt>
<dd><p>Passed to <a href="https://rdrr.io/r/parallel/mclapply.html" class="external-link">parallel::mclapply</a></p></dd>


<dt>alt_col</dt>
<dd><p>Column from <code>var</code> containing alternate bases</p></dd>


<dt>trans_col</dt>
<dd><p>Column from 'exons' containing the transcript_id</p></dd>


<dt>omit_ranges</dt>
<dd><p>GRanges object containing ranges to omit, such as PAR-Y
regions, for example</p></dd>


<dt>tag</dt>
<dd><p>Optional tag to add to all sequence names which were modified</p></dd>


<dt>sep</dt>
<dd><p>Separator to place between seqnames names &amp; tag</p></dd>


<dt>var_tags</dt>
<dd><p>logical(1) Add tags indicating which type of variant were
incorporated, with 's', 'i' and 'd' representing SNPs, Insertions and
Deletions respectively</p></dd>


<dt>var_sep</dt>
<dd><p>Separator between any previous tags and variant tags</p></dd>


<dt>verbose</dt>
<dd><p>logical(1) Include informative messages, or operate silently</p></dd>


<dt>mc.cores</dt>
<dd><p>Number of cores to be used when multi-threading via
<a href="https://rdrr.io/r/parallel/mclapply.html" class="external-link">parallel::mclapply</a></p></dd>


<dt>which</dt>
<dd><p>GRanges object passed to <a href="https://rdrr.io/pkg/VariantAnnotation/man/ScanVcfParam-class.html" class="external-link">VariantAnnotation::ScanVcfParam</a> if
using a VCF directly</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>An XStringSet</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Produce a set of variant modified transcript sequences from a standard
reference genome.
Supported variants are SNPs, Insertions and Deletions</p>
<p>Ranges needing to be masked, such as the Y-chromosome, or Y-PAR can be
provided.</p>
<p><strong>It should be noted that this is a time consuming process</strong>
Inclusion of a large set of insertions and deletions across an entire
transcriptome can involve individually modifying many thousands of
transcripts, which can be a computationally demanding task.
Whilst this can be parallelised using an appropriate number of cores, this
may also prove taxing for lower power laptops, and pre-emptively closing
memory hungry programs such as Slack, or internet browers may be prudent.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicFeatures" class="external-link">GenomicFeatures</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">seq</span> <span class="op">&lt;-</span> <span class="fu">DNAStringSet</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>chr1 <span class="op">=</span> <span class="st">"ACGTAAATGG"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1:1-3:-"</span>, <span class="st">"chr1:7-9:-"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">exons</span><span class="op">$</span><span class="va">transcript_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"trans1"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># When using extractTranscriptSeqs -stranded exons need to be sorted by end</span></span></span>
<span class="r-in"><span><span class="va">exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">exons</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span>, by <span class="op">=</span> <span class="op">~</span><span class="va">end</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">exons</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 2 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand | transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1       7-9      - |        trans1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1       1-3      - |        trans1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span><span class="va">trByExon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">splitAsList</a></span><span class="op">(</span><span class="va">exons</span>, <span class="va">exons</span><span class="op">$</span><span class="va">transcript_id</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Check the sequences</span></span></span>
<span class="r-in"><span><span class="va">seq</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 1:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq                                               names               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    10 ACGTAAATGG                                        chr1</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/extractTranscriptSeqs.html" class="external-link">extractTranscriptSeqs</a></span><span class="op">(</span><span class="va">seq</span>, <span class="va">trByExon</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 1:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq                                               names               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]     6 CATCGT                                            trans1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Define some variants</span></span></span>
<span class="r-in"><span><span class="va">var</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1:2"</span>, <span class="st">"chr1:8"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">var</span><span class="op">$</span><span class="va">ALT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"GGG"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Include the variants adding tags to indicate a SNP and indel</span></span></span>
<span class="r-in"><span><span class="co"># The exons GRanges object will be split by transcript internally</span></span></span>
<span class="r-in"><span><span class="fu">transmogrify</span><span class="op">(</span><span class="va">seq</span>, <span class="va">var</span>, <span class="va">exons</span>, var_tags <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 transcripts found with indels</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 1:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq                                               names               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]     8 CCCCTCTT                                          trans1_si</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://github.com/smped" class="external-link">Stevie Pederson</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

