<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Mogrify a transcriptome using a set of variants — transmogrify • transmogR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Mogrify a transcriptome using a set of variants — transmogrify"><meta name="description" content="Use a set of SNPs, insertions and deletions to modify a reference
transcriptome"><meta property="og:description" content="Use a set of SNPs, insertions and deletions to modify a reference
transcriptome"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="default" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">transmogR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><a class="dropdown-item" href="../articles/creating_a_new_reference.html">Creating a variant modified reference</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/smped/transmogR" aria-label="View on github"><span class="fa fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Mogrify a transcriptome using a set of variants</h1>
      <small class="dont-index">Source: <a href="https://github.com/smped/transmogR/blob/gh-actions/R/transmogrify.R" class="external-link"><code>R/transmogrify.R</code></a></small>
      <div class="d-none name"><code>transmogrify-methods.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Use a set of SNPs, insertions and deletions to modify a reference
transcriptome</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">transmogrify</span><span class="op">(</span><span class="va">x</span>, <span class="va">var</span>, <span class="va">exons</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'XStringSet,GRanges,GRanges'</span></span>
<span><span class="fu">transmogrify</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">var</span>,</span>
<span>  <span class="va">exons</span>,</span>
<span>  alt_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>  trans_col <span class="op">=</span> <span class="st">"transcript_id"</span>,</span>
<span>  omit_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tag <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  var_tags <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  var_sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'BSgenome,GRanges,GRanges'</span></span>
<span><span class="fu">transmogrify</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">var</span>,</span>
<span>  <span class="va">exons</span>,</span>
<span>  alt_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>  trans_col <span class="op">=</span> <span class="st">"transcript_id"</span>,</span>
<span>  omit_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tag <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  var_tags <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  var_sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'BSgenome,VcfFile,GRanges'</span></span>
<span><span class="fu">transmogrify</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">var</span>,</span>
<span>  <span class="va">exons</span>,</span>
<span>  alt_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>  trans_col <span class="op">=</span> <span class="st">"transcript_id"</span>,</span>
<span>  omit_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tag <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  var_tags <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  var_sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">which</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'XStringSet,VcfFile,GRanges'</span></span>
<span><span class="fu">transmogrify</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">var</span>,</span>
<span>  <span class="va">exons</span>,</span>
<span>  alt_col <span class="op">=</span> <span class="st">"ALT"</span>,</span>
<span>  trans_col <span class="op">=</span> <span class="st">"transcript_id"</span>,</span>
<span>  omit_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tag <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  var_tags <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  var_sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">which</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>Reference genome as either a DNAStringSet or BSgenome</p></dd>


<dt id="arg-var">var<a class="anchor" aria-label="anchor" href="#arg-var"></a></dt>
<dd><p>GRanges object containing the variants</p></dd>


<dt id="arg-exons">exons<a class="anchor" aria-label="anchor" href="#arg-exons"></a></dt>
<dd><p>GRanges object with ranges representing exons</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Passed to <a href="https://rdrr.io/r/parallel/mclapply.html" class="external-link">parallel::mclapply</a></p></dd>


<dt id="arg-alt-col">alt_col<a class="anchor" aria-label="anchor" href="#arg-alt-col"></a></dt>
<dd><p>Column from <code>var</code> containing alternate bases</p></dd>


<dt id="arg-trans-col">trans_col<a class="anchor" aria-label="anchor" href="#arg-trans-col"></a></dt>
<dd><p>Column from 'exons' containing the transcript_id</p></dd>


<dt id="arg-omit-ranges">omit_ranges<a class="anchor" aria-label="anchor" href="#arg-omit-ranges"></a></dt>
<dd><p>GRanges object containing ranges to omit, such as PAR-Y
regions, for example</p></dd>


<dt id="arg-tag">tag<a class="anchor" aria-label="anchor" href="#arg-tag"></a></dt>
<dd><p>Optional tag to add to all sequence names which were modified</p></dd>


<dt id="arg-sep">sep<a class="anchor" aria-label="anchor" href="#arg-sep"></a></dt>
<dd><p>Separator to place between seqnames names &amp; tag</p></dd>


<dt id="arg-var-tags">var_tags<a class="anchor" aria-label="anchor" href="#arg-var-tags"></a></dt>
<dd><p>logical(1) Add tags indicating which type of variant were
incorporated, with 's', 'i' and 'd' representing SNPs, Insertions and
Deletions respectively</p></dd>


<dt id="arg-var-sep">var_sep<a class="anchor" aria-label="anchor" href="#arg-var-sep"></a></dt>
<dd><p>Separator between any previous tags and variant tags</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>logical(1) Include informative messages, or operate silently</p></dd>


<dt id="arg-mc-cores">mc.cores<a class="anchor" aria-label="anchor" href="#arg-mc-cores"></a></dt>
<dd><p>Number of cores to be used when multi-threading via
<a href="https://rdrr.io/r/parallel/mclapply.html" class="external-link">parallel::mclapply</a></p></dd>


<dt id="arg-which">which<a class="anchor" aria-label="anchor" href="#arg-which"></a></dt>
<dd><p>GRanges object passed to <a href="https://rdrr.io/pkg/VariantAnnotation/man/ScanVcfParam-class.html" class="external-link">VariantAnnotation::ScanVcfParam</a> if
using a VCF directly</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An XStringSet</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Produce a set of variant modified transcript sequences from a standard
reference genome.
Supported variants are SNPs, Insertions and Deletions</p>
<p>Ranges needing to be masked, such as the Y-chromosome, or Y-PAR can be
provided.</p>
<p><strong>It should be noted that this is a time consuming process</strong>
Inclusion of a large set of insertions and deletions across an entire
transcriptome can involve individually modifying many thousands of
transcripts, which can be a computationally demanding task.
Whilst this can be parallelised using an appropriate number of cores, this
may also prove taxing for lower power laptops, and pre-emptively closing
memory hungry programs such as Slack, or internet browers may be prudent.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicFeatures" class="external-link">GenomicFeatures</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">seq</span> <span class="op">&lt;-</span> <span class="fu">DNAStringSet</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>chr1 <span class="op">=</span> <span class="st">"ACGTAAATGG"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1:1-3:-"</span>, <span class="st">"chr1:7-9:-"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">exons</span><span class="op">$</span><span class="va">transcript_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"trans1"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># When using extractTranscriptSeqs -stranded exons need to be sorted by end</span></span></span>
<span class="r-in"><span><span class="va">exons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">exons</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span>, by <span class="op">=</span> <span class="op">~</span><span class="va">end</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">exons</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 2 ranges and 1 metadata column:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames    ranges strand | transcript_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     chr1       7-9      - |        trans1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]     chr1       1-3      - |        trans1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-in"><span><span class="va">trByExon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">splitAsList</a></span><span class="op">(</span><span class="va">exons</span>, <span class="va">exons</span><span class="op">$</span><span class="va">transcript_id</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Check the sequences</span></span></span>
<span class="r-in"><span><span class="va">seq</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 1:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq                                               names               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    10 ACGTAAATGG                                        chr1</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/extractTranscriptSeqs.html" class="external-link">extractTranscriptSeqs</a></span><span class="op">(</span><span class="va">seq</span>, <span class="va">trByExon</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 1:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq                                               names               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]     6 CATCGT                                            trans1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Define some variants</span></span></span>
<span class="r-in"><span><span class="va">var</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1:2"</span>, <span class="st">"chr1:8"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">var</span><span class="op">$</span><span class="va">ALT</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"GGG"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Include the variants adding tags to indicate a SNP and indel</span></span></span>
<span class="r-in"><span><span class="co"># The exons GRanges object will be split by transcript internally</span></span></span>
<span class="r-in"><span><span class="fu">transmogrify</span><span class="op">(</span><span class="va">seq</span>, <span class="va">var</span>, <span class="va">exons</span>, var_tags <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 transcripts found with indels</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 1:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq                                               names               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]     8 CCCCTCTT                                          trans1_si</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/smped" class="external-link">Stevie Pederson</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer></div>





  </body></html>

