<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Creating a Variant-Modified Reference • transmogR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Creating a Variant-Modified Reference">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="default" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">transmogR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/creating_a_new_reference.html">Creating a variant modified reference</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smped/transmogR" aria-label="View on github"><span class="fa fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Creating a Variant-Modified Reference</h1>
                        <h4 data-toc-skip class="author">Stevie
Pederson</h4>
            <address class="author_afil">
      Black Ochre Data Labs, Telethon Kid’s Institute, Adelaide,
Australia<br><a class="author_email" href="mailto:#"></a><a href="mailto:stephen.pederson@telethonkids.org.au" class="email">stephen.pederson@telethonkids.org.au</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/smped/transmogR/blob/gh-actions/vignettes/creating_a_new_reference.Rmd" class="external-link"><code>vignettes/creating_a_new_reference.Rmd</code></a></small>
      <div class="d-none name"><code>creating_a_new_reference.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The use of personalised or population-level variants opens the
      door to the possibility of creating a variant-modified reference
      genome, or transcriptome. The package transmogR allows the
      creation of both, with a focus on combining both in order to
      created a custom reference transcriptome, along with decoy
      transcripts for use with the pseudo-aligner salmon.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The incorporation of personalised or population-level variants into a
<em>reference genome</em> has been shown to have a significant impact on
subsequent alignments <span class="citation">(Kaminow et al.
2022)</span>. Whilst implemented for splice-aware alignment or RNA-Seq
data using <em>STARconsensus</em>, the package <code>transmogR</code>
enables the creation of a variant-modified <em>reference
transcriptome</em> for use with pseudo aligners such as <em>salmon</em>
<span class="citation">(Srivastava et al. 2020)</span>. In addition,
multiple visualisation and summarisation methods are included for a
cursory analysis of any custom variant sets being used.</p>
<p>Whilst the subsequent code is demonstrated on a small genomic region,
the complete process for creating a modified a reference can run in
under 20 minutes if using 4 or more cores.</p>
<p>Importantly, it is expected that the user will have carefully
prepared their set of variants such that only the following exact
variant types are included.</p>
<ul>
<li>
<strong>SNV</strong>/<strong>SNP</strong>: The associated range must
be of width 1 with a single base as the replacement allele</li>
<li>
<strong>Insertions</strong>: The position being replaced must be a
single nucleotide with the replacement bases being &gt; 1nt</li>
<li>
<strong>Deletions</strong>: The position being replaced must be &gt;
1nt, whilst the replacement allele must be a single base</li>
</ul>
<p>Any variants which do not conform to these criteria will likely cause
a series of frustrating errors when attempting to use this package.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>In order to perform the operations in this vignette, the following
packages require installation.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"BiocManager"</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"transmogR"</span><span class="op">)</span></span></code></pre></div>
<p>Once these packages are installed, we can load them easily</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">VariantAnnotation</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/smped/extraChIPs" class="external-link">extraChIPs</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/smped/transmogR" class="external-link">transmogR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-data">Required Data<a class="anchor" aria-label="anchor" href="#required-data"></a>
</h3>
<p>In order to create a modified reference, three primary data objects
are required: 1) a reference genome; 2) a set of genomic variants; and
3) a set of exon-level co-ordinates defining transcript structure.</p>
<p>For this vignette, we’ll use GRCh38 as our primary reference genome,
but restricting the sequences to <em>chr1</em> only. The package can
take either a <code>DNAStringSet</code> or <code>BSgenome</code> object
as the reference genome.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html" class="external-link">getSeq</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span>, <span class="st">"chr1"</span><span class="op">)</span></span>
<span><span class="va">chr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">chr1</span>, <span class="st">"DNAStringSet"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">chr1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"chr1"</span></span>
<span><span class="va">chr1</span></span></code></pre></div>
<pre><code><span><span class="co">## DNAStringSet object of length 1:</span></span>
<span><span class="co">##         width seq                                           names               </span></span>
<span><span class="co">## [1] 248956422 <span style="color: #EEEEEE; background-color: #B2B2B2;">NNNNNNNNNNNNNNNNNNNNN</span>...<span style="color: #EEEEEE; background-color: #B2B2B2;">NNNNNNNNNNNNNNNNNNNNN</span> chr1</span></span></code></pre>
<p>A small set of variants has been provided with the package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">chr1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">sq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"GRCh38"</span></span>
<span><span class="va">vcf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/1000GP_subset.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"transmogR"</span><span class="op">)</span></span>
<span><span class="va">vcf_param</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/ScanVcfParam-class.html" class="external-link">ScanVcfParam</a></span><span class="op">(</span>fixed <span class="op">=</span> <span class="st">"ALT"</span>, info <span class="op">=</span> <span class="cn">NA</span>, which <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">sq</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">var</span> <span class="op">&lt;-</span> <span class="fu">rowRanges</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/readVcf-methods.html" class="external-link">readVcf</a></span><span class="op">(</span><span class="va">vcf</span>, param <span class="op">=</span> <span class="va">vcf_param</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">var</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sq</span></span>
<span><span class="va">var</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 100 ranges and 3 metadata columns:</span></span>
<span><span class="co">##                         seqnames        ranges strand | paramRangeID</span></span>
<span><span class="co">##                            &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;factor&gt;</span></span>
<span><span class="co">##            1:113969:C:T     chr1        113969      * |         chr1</span></span>
<span><span class="co">##            1:121009:C:T     chr1        121009      * |         chr1</span></span>
<span><span class="co">##            1:123511:G:A     chr1        123511      * |         chr1</span></span>
<span><span class="co">##            1:126113:C:A     chr1        126113      * |         chr1</span></span>
<span><span class="co">##            1:128798:C:T     chr1        128798      * |         chr1</span></span>
<span><span class="co">##                     ...      ...           ...    ... .          ...</span></span>
<span><span class="co">##            1:839405:G:A     chr1        839405      * |         chr1</span></span>
<span><span class="co">##   1:839480:CACACACCTG:C     chr1 839480-839494      * |         chr1</span></span>
<span><span class="co">##   1:839515:CTAGACACAC:C     chr1 839515-839543      * |         chr1</span></span>
<span><span class="co">##            1:840046:G:A     chr1        840046      * |         chr1</span></span>
<span><span class="co">##            1:840279:A:G     chr1        840279      * |         chr1</span></span>
<span><span class="co">##                                             REF                ALT</span></span>
<span><span class="co">##                                  &lt;DNAStringSet&gt; &lt;DNAStringSetList&gt;</span></span>
<span><span class="co">##            1:113969:C:T                       C                  T</span></span>
<span><span class="co">##            1:121009:C:T                       C                  T</span></span>
<span><span class="co">##            1:123511:G:A                       G                  A</span></span>
<span><span class="co">##            1:126113:C:A                       C                  A</span></span>
<span><span class="co">##            1:128798:C:T                       C                  T</span></span>
<span><span class="co">##                     ...                     ...                ...</span></span>
<span><span class="co">##            1:839405:G:A                       G                  A</span></span>
<span><span class="co">##   1:839480:CACACACCTG:C         CACACACCTGGACAA                  C</span></span>
<span><span class="co">##   1:839515:CTAGACACAC:C CTAGACACAC...CACACACACG                  C</span></span>
<span><span class="co">##            1:840046:G:A                       G                  A</span></span>
<span><span class="co">##            1:840279:A:G                       A                  G</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from GRCh38 genome</span></span></code></pre>
<p>An additional set of transcripts derived from Gencode v44<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="https://www.gencodegenes.org/human/" class="external-link uri"&gt;https://www.gencodegenes.org/human/&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a> has also
been provided.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/gencode.v44.subset.gtf.gz"</span>, package <span class="op">=</span> <span class="st">"transmogR"</span><span class="op">)</span></span>
<span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/GFFFile-class.html" class="external-link">import.gff</a></span><span class="op">(</span><span class="va">f</span>, which <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="va">sq</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sq</span></span>
<span><span class="va">gtf</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 603 ranges and 11 metadata columns:</span></span>
<span><span class="co">##         seqnames        ranges strand |      source       type     score</span></span>
<span><span class="co">##            &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |    &lt;factor&gt;   &lt;factor&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##     [1]     chr1  89295-133723      - | rtracklayer gene              NA</span></span>
<span><span class="co">##     [2]     chr1  89295-120932      - | rtracklayer transcript        NA</span></span>
<span><span class="co">##     [3]     chr1 120775-120932      - | rtracklayer exon              NA</span></span>
<span><span class="co">##     [4]     chr1 112700-112804      - | rtracklayer exon              NA</span></span>
<span><span class="co">##     [5]     chr1   92091-92240      - | rtracklayer exon              NA</span></span>
<span><span class="co">##     ...      ...           ...    ... .         ...        ...       ...</span></span>
<span><span class="co">##   [599]     chr1 852671-852766      + | rtracklayer exon              NA</span></span>
<span><span class="co">##   [600]     chr1 853391-854096      + | rtracklayer exon              NA</span></span>
<span><span class="co">##   [601]     chr1 851348-852752      + | rtracklayer transcript        NA</span></span>
<span><span class="co">##   [602]     chr1 851348-852110      + | rtracklayer exon              NA</span></span>
<span><span class="co">##   [603]     chr1 852671-852752      + | rtracklayer exon              NA</span></span>
<span><span class="co">##             phase            gene_id   gene_type       gene_name</span></span>
<span><span class="co">##         &lt;integer&gt;        &lt;character&gt; &lt;character&gt;     &lt;character&gt;</span></span>
<span><span class="co">##     [1]      &lt;NA&gt;  ENSG00000238009.6      lncRNA ENSG00000238009</span></span>
<span><span class="co">##     [2]      &lt;NA&gt;  ENSG00000238009.6      lncRNA ENSG00000238009</span></span>
<span><span class="co">##     [3]      &lt;NA&gt;  ENSG00000238009.6      lncRNA ENSG00000238009</span></span>
<span><span class="co">##     [4]      &lt;NA&gt;  ENSG00000238009.6      lncRNA ENSG00000238009</span></span>
<span><span class="co">##     [5]      &lt;NA&gt;  ENSG00000238009.6      lncRNA ENSG00000238009</span></span>
<span><span class="co">##     ...       ...                ...         ...             ...</span></span>
<span><span class="co">##   [599]      &lt;NA&gt; ENSG00000228794.12      lncRNA       LINC01128</span></span>
<span><span class="co">##   [600]      &lt;NA&gt; ENSG00000228794.12      lncRNA       LINC01128</span></span>
<span><span class="co">##   [601]      &lt;NA&gt; ENSG00000228794.12      lncRNA       LINC01128</span></span>
<span><span class="co">##   [602]      &lt;NA&gt; ENSG00000228794.12      lncRNA       LINC01128</span></span>
<span><span class="co">##   [603]      &lt;NA&gt; ENSG00000228794.12      lncRNA       LINC01128</span></span>
<span><span class="co">##             transcript_id transcript_type transcript_name exon_number</span></span>
<span><span class="co">##               &lt;character&gt;     &lt;character&gt;     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">##     [1]              &lt;NA&gt;            &lt;NA&gt;            &lt;NA&gt;        &lt;NA&gt;</span></span>
<span><span class="co">##     [2] ENST00000466430.5          lncRNA ENST00000466430        &lt;NA&gt;</span></span>
<span><span class="co">##     [3] ENST00000466430.5          lncRNA ENST00000466430           1</span></span>
<span><span class="co">##     [4] ENST00000466430.5          lncRNA ENST00000466430           2</span></span>
<span><span class="co">##     [5] ENST00000466430.5          lncRNA ENST00000466430           3</span></span>
<span><span class="co">##     ...               ...             ...             ...         ...</span></span>
<span><span class="co">##   [599] ENST00000688420.1          lncRNA   LINC01128-235           4</span></span>
<span><span class="co">##   [600] ENST00000688420.1          lncRNA   LINC01128-235           5</span></span>
<span><span class="co">##   [601] ENST00000425657.1 retained_intron   LINC01128-203        &lt;NA&gt;</span></span>
<span><span class="co">##   [602] ENST00000425657.1 retained_intron   LINC01128-203           1</span></span>
<span><span class="co">##   [603] ENST00000425657.1 retained_intron   LINC01128-203           2</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from GRCh38 genome</span></span></code></pre>
<p>Splitting this gtf into feature types can also be very handy for
downstream processes.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/splitAsList.html" class="external-link">splitAsList</a></span><span class="op">(</span><span class="va">gtf</span>, <span class="va">gtf</span><span class="op">$</span><span class="va">type</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="inspecting-variants">Inspecting Variants<a class="anchor" aria-label="anchor" href="#inspecting-variants"></a>
</h2>
<p>Knowing where our variants lie, and how they relate to each other can
be informative, and as such, some simple visualisation and summarisation
functions have been included. In the following, we can check to see how
many exons directly overlap a variant, showing how many unique genes
this summarises to. Any ids, which don’t overlap a variants are also
described in the plot title.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/upsetVarByCol.html">upsetVarByCol</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">$</span><span class="va">exon</span>, <span class="va">var</span>, mcol <span class="op">=</span> <span class="st">"gene_id"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="creating_a_new_reference_files/figure-html/upset-var-1.png" alt="Included variants which overlap exonic sequences, summarised by unique gene ids" width="700"><p class="caption">
Included variants which overlap exonic sequences, summarised by unique
gene ids
</p>
</div>
<p>In addition, we can obtain a simple breakdown of overlapping regions
using a GRangesList. We can use the function
<code><a href="https://rdrr.io/pkg/extraChIPs/man/defineRegions.html" class="external-link">defineRegions()</a></code> from <code>extraChIPs</code> to define
regions based on gene &amp; transcript locations. This function assigns
each position in the genome uniquely to a given feature hierarchically,
using all provided transcripts, meaning some exons will be considered as
promoters. To ensure that all exons are included as exons, we can just
substitute in the values from our gtf for this feature type.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/extraChIPs/man/defineRegions.html" class="external-link">defineRegions</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">$</span><span class="va">gene</span>, <span class="va">gtf</span><span class="op">$</span><span class="va">transcript</span>, <span class="va">gtf</span><span class="op">$</span><span class="va">exon</span>, proximal <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">regions</span><span class="op">$</span><span class="va">exon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">$</span><span class="va">exon</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/overlapsByVar-methods.html">overlapsByVar</a></span><span class="op">(</span><span class="va">regions</span>, <span class="va">var</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   Deletion Insertion SNV Total</span></span>
<span><span class="co">## promoter                 1         1  46    48</span></span>
<span><span class="co">## upstream_promoter        0         0  20    20</span></span>
<span><span class="co">## exon                     0         1  37    38</span></span>
<span><span class="co">## intron                   5         0  24    29</span></span>
<span><span class="co">## intergenic               0         0   0     0</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="creating-modified-reference-sequences">Creating Modified Reference Sequences<a class="anchor" aria-label="anchor" href="#creating-modified-reference-sequences"></a>
</h2>
<div class="section level3">
<h3 id="modifying-a-reference-genome">Modifying a Reference Genome<a class="anchor" aria-label="anchor" href="#modifying-a-reference-genome"></a>
</h3>
<p>The simplest method for modifying a reference genome is to simply
call <code><a href="../reference/genomogrify-methods.html">genomogrify()</a></code> with either a <code>DNAStringSet</code>
or <code>BSgenome</code> object. A tag can be optionally added to all
sequence names to avoid confusion.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chr1_mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genomogrify-methods.html">genomogrify</a></span><span class="op">(</span><span class="va">chr1</span>, <span class="va">var</span>, tag <span class="op">=</span> <span class="st">"mod"</span><span class="op">)</span></span>
<span><span class="va">chr1_mod</span></span></code></pre></div>
<pre><code><span><span class="co">## DNAStringSet object of length 1:</span></span>
<span><span class="co">##         width seq                                           names               </span></span>
<span><span class="co">## [1] 248956362 <span style="color: #EEEEEE; background-color: #B2B2B2;">NNNNNNNNNNNNNNNNNNNNN</span>...<span style="color: #EEEEEE; background-color: #B2B2B2;">NNNNNNNNNNNNNNNNNNNNN</span> chr1_mod</span></span></code></pre>
<p>The new reference genome can be exported to fasta format using
<code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">writeXStringSet()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="modifying-a-reference-transcriptome">Modifying a Reference Transcriptome<a class="anchor" aria-label="anchor" href="#modifying-a-reference-transcriptome"></a>
</h3>
<p>The process for creating a variant-modified reference transcriptome
is essentially the same as above, with the addition of exon-level
co-ordinates for the set of transcripts. An optional tag can be added to
all transcripts to indicate which have been modified, with an additional
tag able to be included which indicates which type of variant has been
incorporated into the new transcript sequence. Variant tags will be one
of <code>s</code>, <code>i</code> or <code>d</code> to indicate SNV,
Insertions or Deletions respectively. In our example dataset, only one
transcript contains both SNVs and an insertion.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trans_mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transmogrify-methods.html">transmogrify</a></span><span class="op">(</span></span>
<span>    <span class="va">chr1</span>, <span class="va">var</span>, <span class="va">gtf</span><span class="op">$</span><span class="va">exon</span>, tag <span class="op">=</span> <span class="st">"mod"</span>, var_tags <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">trans_mod</span></span></code></pre></div>
<pre><code><span><span class="co">## DNAStringSet object of length 112:</span></span>
<span><span class="co">##       width seq                                             names               </span></span>
<span><span class="co">##   [1]  1947 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>...<span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span> ENST00000326734.2...</span></span>
<span><span class="co">##   [2]  1702 <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span>...<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span> ENST00000357876.6</span></span>
<span><span class="co">##   [3]  1358 <span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GGGGG</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span> ENST00000412115.2...</span></span>
<span><span class="co">##   [4]   421 <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span> ENST00000414688.6</span></span>
<span><span class="co">##   [5]   894 <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span>...<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span> ENST00000415295.1</span></span>
<span><span class="co">##   ...   ... ...</span></span>
<span><span class="co">## [108]  3097 <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">AA</span>...<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAAAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">CCCC</span> ENST00000701768.1</span></span>
<span><span class="co">## [109]  1601 <span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span> ENST00000702098.1</span></span>
<span><span class="co">## [110]  1432 <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFAFAF;">A</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAAA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span> ENST00000702273.1</span></span>
<span><span class="co">## [111]  1460 <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAAAA</span><span style="color: #000000; background-color: #FFD7AF;">TTT</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span> ENST00000702557.1</span></span>
<span><span class="co">## [112]  1651 <span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span>...<span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">AAAA</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GG</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span> ENST00000702847.1</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">trans_mod</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"_si"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">trans_mod</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ENST00000473798.2_mod_si"</span></span></code></pre>
<p>This can be simply exported again using
<code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">writeXStringSet()</a></code>. If using decoy transcripts for
<code>salmon</code>, the <code>gentrome</code> can also be simply
exported by combining the two modified references.</p>
<p>Both of the above processes rely on the lower-level functions
<code><a href="../reference/owl-methods.html">owl()</a></code> and <code><a href="../reference/indelcator-methods.html">indelcator()</a></code> which <em>overwrite
letters</em> or substitute <em>indels</em> respectively. These are also
able to be called individually as shown in the help pages.</p>
</div>
</div>
<div class="section level2">
<h2 id="additional-capabilities">Additional Capabilities<a class="anchor" aria-label="anchor" href="#additional-capabilities"></a>
</h2>
<div class="section level3">
<h3 id="pseudo-autosomal-regions-par-y">Pseudo-Autosomal Regions (PAR-Y)<a class="anchor" aria-label="anchor" href="#pseudo-autosomal-regions-par-y"></a>
</h3>
<p>Beyond these lower-level functions, PAR-Y regions for hg38, hg19 and
CHM13 are able to obtained using <code><a href="../reference/parY-methods.html">parY()</a></code> and passing the
appropriate <code>seqinfo</code> object. This <code>seqinfo</code>
object checks the length of the Y-chromosome and guesses which reference
genome has been used.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sq_hg38</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/parY-methods.html">parY</a></span><span class="op">(</span><span class="va">sq_hg38</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 2 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames            ranges strand |         PAR</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">##   [1]     chrY     10001-2781479      * |        PAR1</span></span>
<span><span class="co">##   [2]     chrY 56887903-57217415      * |        PAR2</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 711 sequences (1 circular) from hg38 genome</span></span></code></pre>
<p>If the user wishes to exclude transcripts in the PAR-Y region, these
ranges can be passed to <code><a href="../reference/transmogrify-methods.html">transmogrify()</a></code> and any transcripts
which overlap the PAR-Y region will be excluded. Alternatively, passing
the entire Y-chromosome to <code><a href="../reference/transmogrify-methods.html">transmogrify()</a></code> will exclude all
transcripts in the Y-chromosome, as may be preferred for female-specific
references.</p>
<p>These regions can also be passed to <code><a href="../reference/genomogrify-methods.html">genomogrify()</a></code> as a
mask, with all bases within the masked region being re-assigned as an
N.</p>
</div>
<div class="section level3">
<h3 id="splice-junctions">Splice Junctions<a class="anchor" aria-label="anchor" href="#splice-junctions"></a>
</h3>
<p>In addition, the set of splice junctions associated with any
transcript can be obtained using our known exons.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"transcript_id"</span>, <span class="st">"transcript_name"</span>, <span class="st">"gene_id"</span>, <span class="st">"gene_name"</span><span class="op">)</span></span>
<span><span class="va">sj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sjFromExons.html">sjFromExons</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">$</span><span class="va">exon</span>, extra_cols <span class="op">=</span> <span class="va">ec</span><span class="op">)</span></span>
<span><span class="va">sj</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 730 ranges and 6 metadata columns:</span></span>
<span><span class="co">##         seqnames        ranges strand |        site     transcript_id</span></span>
<span><span class="co">##            &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;       &lt;character&gt;</span></span>
<span><span class="co">##     [1]     chr1 182745-182752      + |       donor ENST00000624431.2</span></span>
<span><span class="co">##     [2]     chr1 183128-183132      + |    acceptor ENST00000624431.2</span></span>
<span><span class="co">##     [3]     chr1 183215-183222      + |       donor ENST00000624431.2</span></span>
<span><span class="co">##     [4]     chr1 183490-183494      + |    acceptor ENST00000624431.2</span></span>
<span><span class="co">##     [5]     chr1 183570-183577      + |       donor ENST00000624431.2</span></span>
<span><span class="co">##     ...      ...           ...    ... .         ...               ...</span></span>
<span><span class="co">##   [726]     chr1 810061-810068      - |       donor ENST00000590817.3</span></span>
<span><span class="co">##   [727]     chr1 810170-810174      - |    acceptor ENST00000447500.4</span></span>
<span><span class="co">##   [728]     chr1 817367-817374      - |       donor ENST00000447500.4</span></span>
<span><span class="co">##   [729]     chr1 827664-827671      - |       donor ENST00000635509.2</span></span>
<span><span class="co">##   [730]     chr1 827664-827671      - |       donor ENST00000634337.2</span></span>
<span><span class="co">##         transcript_name            gene_id       gene_name exon_number</span></span>
<span><span class="co">##             &lt;character&gt;        &lt;character&gt;     &lt;character&gt;   &lt;integer&gt;</span></span>
<span><span class="co">##     [1]    DDX11L17-201  ENSG00000279928.2        DDX11L17           1</span></span>
<span><span class="co">##     [2]    DDX11L17-201  ENSG00000279928.2        DDX11L17           2</span></span>
<span><span class="co">##     [3]    DDX11L17-201  ENSG00000279928.2        DDX11L17           2</span></span>
<span><span class="co">##     [4]    DDX11L17-201  ENSG00000279928.2        DDX11L17           3</span></span>
<span><span class="co">##     [5]    DDX11L17-201  ENSG00000279928.2        DDX11L17           3</span></span>
<span><span class="co">##     ...             ...                ...             ...         ...</span></span>
<span><span class="co">##   [726] ENST00000590817  ENSG00000230092.8 ENSG00000230092           1</span></span>
<span><span class="co">##   [727] ENST00000447500  ENSG00000290784.1 ENSG00000290784           2</span></span>
<span><span class="co">##   [728] ENST00000447500  ENSG00000290784.1 ENSG00000290784           1</span></span>
<span><span class="co">##   [729] ENST00000635509 ENSG00000230021.10 ENSG00000230021           1</span></span>
<span><span class="co">##   [730] ENST00000634337 ENSG00000230021.10 ENSG00000230021           1</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from GRCh38 genome</span></span></code></pre>
<p>Many splice junctions will be shared across multiple transcripts, so
the returned set of junctions can also be simplified using
<code><a href="https://rdrr.io/pkg/extraChIPs/man/chopMC.html" class="external-link">chopMC()</a></code> from <code>extraChIPs</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/extraChIPs/man/chopMC.html" class="external-link">chopMC</a></span><span class="op">(</span><span class="va">sj</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 196 ranges and 6 metadata columns:</span></span>
<span><span class="co">##         seqnames        ranges strand |        site</span></span>
<span><span class="co">##            &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">##     [1]     chr1 182745-182752      + |       donor</span></span>
<span><span class="co">##     [2]     chr1 183128-183132      + |    acceptor</span></span>
<span><span class="co">##     [3]     chr1 183215-183222      + |       donor</span></span>
<span><span class="co">##     [4]     chr1 183490-183494      + |    acceptor</span></span>
<span><span class="co">##     [5]     chr1 183570-183577      + |       donor</span></span>
<span><span class="co">##     ...      ...           ...    ... .         ...</span></span>
<span><span class="co">##   [192]     chr1 808623-808627      - |    acceptor</span></span>
<span><span class="co">##   [193]     chr1 810061-810068      - |       donor</span></span>
<span><span class="co">##   [194]     chr1 810170-810174      - |    acceptor</span></span>
<span><span class="co">##   [195]     chr1 817367-817374      - |       donor</span></span>
<span><span class="co">##   [196]     chr1 827664-827671      - |       donor</span></span>
<span><span class="co">##                               transcript_id                 transcript_name</span></span>
<span><span class="co">##                             &lt;CharacterList&gt;                 &lt;CharacterList&gt;</span></span>
<span><span class="co">##     [1]                   ENST00000624431.2                    DDX11L17-201</span></span>
<span><span class="co">##     [2]                   ENST00000624431.2                    DDX11L17-201</span></span>
<span><span class="co">##     [3]                   ENST00000624431.2                    DDX11L17-201</span></span>
<span><span class="co">##     [4]                   ENST00000624431.2                    DDX11L17-201</span></span>
<span><span class="co">##     [5]                   ENST00000624431.2                    DDX11L17-201</span></span>
<span><span class="co">##     ...                                 ...                             ...</span></span>
<span><span class="co">##   [192] ENST00000447500.4,ENST00000590817.3 ENST00000447500,ENST00000590817</span></span>
<span><span class="co">##   [193] ENST00000447500.4,ENST00000590817.3 ENST00000447500,ENST00000590817</span></span>
<span><span class="co">##   [194]                   ENST00000447500.4                 ENST00000447500</span></span>
<span><span class="co">##   [195]                   ENST00000447500.4                 ENST00000447500</span></span>
<span><span class="co">##   [196] ENST00000635509.2,ENST00000634337.2 ENST00000635509,ENST00000634337</span></span>
<span><span class="co">##                                       gene_id                       gene_name</span></span>
<span><span class="co">##                               &lt;CharacterList&gt;                 &lt;CharacterList&gt;</span></span>
<span><span class="co">##     [1]                     ENSG00000279928.2                        DDX11L17</span></span>
<span><span class="co">##     [2]                     ENSG00000279928.2                        DDX11L17</span></span>
<span><span class="co">##     [3]                     ENSG00000279928.2                        DDX11L17</span></span>
<span><span class="co">##     [4]                     ENSG00000279928.2                        DDX11L17</span></span>
<span><span class="co">##     [5]                     ENSG00000279928.2                        DDX11L17</span></span>
<span><span class="co">##     ...                                   ...                             ...</span></span>
<span><span class="co">##   [192]   ENSG00000290784.1,ENSG00000230092.8 ENSG00000290784,ENSG00000230092</span></span>
<span><span class="co">##   [193]   ENSG00000290784.1,ENSG00000230092.8 ENSG00000290784,ENSG00000230092</span></span>
<span><span class="co">##   [194]                     ENSG00000290784.1                 ENSG00000290784</span></span>
<span><span class="co">##   [195]                     ENSG00000290784.1                 ENSG00000290784</span></span>
<span><span class="co">##   [196] ENSG00000230021.10,ENSG00000230021.10 ENSG00000230021,ENSG00000230021</span></span>
<span><span class="co">##           exon_number</span></span>
<span><span class="co">##         &lt;IntegerList&gt;</span></span>
<span><span class="co">##     [1]             1</span></span>
<span><span class="co">##     [2]             2</span></span>
<span><span class="co">##     [3]             2</span></span>
<span><span class="co">##     [4]             3</span></span>
<span><span class="co">##     [5]             3</span></span>
<span><span class="co">##     ...           ...</span></span>
<span><span class="co">##   [192]           3,2</span></span>
<span><span class="co">##   [193]           2,1</span></span>
<span><span class="co">##   [194]             2</span></span>
<span><span class="co">##   [195]             1</span></span>
<span><span class="co">##   [196]           1,1</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>As a further alternative, splice junctions can be returned as a set
of interactions, with donor sites being assigned to the
<code>anchorOne</code> element, and acceptor sites being placed in the
<code>anchorTwo</code> element This enables the identification of all
splice junctions for specific transcripts.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sjFromExons.html">sjFromExons</a></span><span class="op">(</span><span class="va">gtf</span><span class="op">$</span><span class="va">exon</span>, extra_cols <span class="op">=</span> <span class="va">ec</span>, as <span class="op">=</span> <span class="st">"GInteractions"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">sj</span>, <span class="va">transcript_name</span> <span class="op">==</span> <span class="st">"DDX11L17-201"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GInteractions object with 4 interactions and 5 metadata columns:</span></span>
<span><span class="co">##       seqnames1       ranges1     seqnames2       ranges2 |     transcript_id</span></span>
<span><span class="co">##           &lt;Rle&gt;     &lt;IRanges&gt;         &lt;Rle&gt;     &lt;IRanges&gt; |       &lt;character&gt;</span></span>
<span><span class="co">##   [1]      chr1 182745-182752 ---      chr1 183128-183132 | ENST00000624431.2</span></span>
<span><span class="co">##   [2]      chr1 183215-183222 ---      chr1 183490-183494 | ENST00000624431.2</span></span>
<span><span class="co">##   [3]      chr1 183570-183577 ---      chr1 183736-183740 | ENST00000624431.2</span></span>
<span><span class="co">##   [4]      chr1 183900-183907 ---      chr1 183977-183981 | ENST00000624431.2</span></span>
<span><span class="co">##       transcript_name           gene_id   gene_name        sj</span></span>
<span><span class="co">##           &lt;character&gt;       &lt;character&gt; &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">##   [1]    DDX11L17-201 ENSG00000279928.2    DDX11L17         1</span></span>
<span><span class="co">##   [2]    DDX11L17-201 ENSG00000279928.2    DDX11L17         2</span></span>
<span><span class="co">##   [3]    DDX11L17-201 ENSG00000279928.2    DDX11L17         3</span></span>
<span><span class="co">##   [4]    DDX11L17-201 ENSG00000279928.2    DDX11L17         4</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   regions: 196 ranges and 0 metadata columns</span></span>
<span><span class="co">##   seqinfo: 1 sequence from GRCh38 genome</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">## R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] BSgenome.Hsapiens.UCSC.hg38_1.4.5 BSgenome_1.73.1                  </span></span>
<span><span class="co">##  [3] BiocIO_1.15.2                     transmogR_1.1.3                  </span></span>
<span><span class="co">##  [5] extraChIPs_1.9.6                  tibble_3.2.1                     </span></span>
<span><span class="co">##  [7] ggside_0.3.1                      ggplot2_3.5.1                    </span></span>
<span><span class="co">##  [9] BiocParallel_1.39.0               rtracklayer_1.65.0               </span></span>
<span><span class="co">## [11] VariantAnnotation_1.51.1          Rsamtools_2.21.2                 </span></span>
<span><span class="co">## [13] Biostrings_2.73.2                 XVector_0.45.0                   </span></span>
<span><span class="co">## [15] SummarizedExperiment_1.35.4       Biobase_2.65.1                   </span></span>
<span><span class="co">## [17] GenomicRanges_1.57.2              GenomeInfoDb_1.41.2              </span></span>
<span><span class="co">## [19] IRanges_2.39.2                    S4Vectors_0.43.2                 </span></span>
<span><span class="co">## [21] MatrixGenerics_1.17.0             matrixStats_1.4.1                </span></span>
<span><span class="co">## [23] BiocGenerics_0.51.3               BiocStyle_2.33.1                 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] bitops_1.0-9               filelock_1.0.3            </span></span>
<span><span class="co">##   [3] polyclip_1.10-7            XML_3.99-0.17             </span></span>
<span><span class="co">##   [5] rpart_4.1.23               lifecycle_1.0.4           </span></span>
<span><span class="co">##   [7] httr2_1.0.5                edgeR_4.3.19              </span></span>
<span><span class="co">##   [9] MASS_7.3-61                lattice_0.22-6            </span></span>
<span><span class="co">##  [11] ensembldb_2.29.1           backports_1.5.0           </span></span>
<span><span class="co">##  [13] magrittr_2.0.3             limma_3.61.12             </span></span>
<span><span class="co">##  [15] Hmisc_5.1-3                sass_0.4.9                </span></span>
<span><span class="co">##  [17] rmarkdown_2.28             jquerylib_0.1.4           </span></span>
<span><span class="co">##  [19] yaml_2.3.10                metapod_1.13.0            </span></span>
<span><span class="co">##  [21] Gviz_1.49.0                DBI_1.2.3                 </span></span>
<span><span class="co">##  [23] RColorBrewer_1.1-3         abind_1.4-8               </span></span>
<span><span class="co">##  [25] zlibbioc_1.51.1            purrr_1.0.2               </span></span>
<span><span class="co">##  [27] AnnotationFilter_1.29.0    biovizBase_1.53.0         </span></span>
<span><span class="co">##  [29] RCurl_1.98-1.16            nnet_7.3-19               </span></span>
<span><span class="co">##  [31] tweenr_2.0.3               rappdirs_0.3.3            </span></span>
<span><span class="co">##  [33] GenomeInfoDbData_1.2.13    ggrepel_0.9.6             </span></span>
<span><span class="co">##  [35] pkgdown_2.1.1.9000         codetools_0.2-20          </span></span>
<span><span class="co">##  [37] DelayedArray_0.31.14       ggforce_0.4.2             </span></span>
<span><span class="co">##  [39] xml2_1.3.6                 tidyselect_1.2.1          </span></span>
<span><span class="co">##  [41] futile.logger_1.4.3        UCSC.utils_1.1.0          </span></span>
<span><span class="co">##  [43] farver_2.1.2               ComplexUpset_1.3.3        </span></span>
<span><span class="co">##  [45] BiocFileCache_2.13.2       base64enc_0.1-3           </span></span>
<span><span class="co">##  [47] GenomicAlignments_1.41.0   jsonlite_1.8.9            </span></span>
<span><span class="co">##  [49] Formula_1.2-5              systemfonts_1.1.0         </span></span>
<span><span class="co">##  [51] tools_4.4.1                progress_1.2.3            </span></span>
<span><span class="co">##  [53] ragg_1.3.3                 Rcpp_1.0.13               </span></span>
<span><span class="co">##  [55] glue_1.8.0                 gridExtra_2.3             </span></span>
<span><span class="co">##  [57] SparseArray_1.5.45         xfun_0.48                 </span></span>
<span><span class="co">##  [59] dplyr_1.1.4                withr_3.0.1               </span></span>
<span><span class="co">##  [61] formatR_1.14               BiocManager_1.30.25       </span></span>
<span><span class="co">##  [63] fastmap_1.2.0              latticeExtra_0.6-30       </span></span>
<span><span class="co">##  [65] fansi_1.0.6                digest_0.6.37             </span></span>
<span><span class="co">##  [67] R6_2.5.1                   textshaping_0.4.0         </span></span>
<span><span class="co">##  [69] colorspace_2.1-1           jpeg_0.1-10               </span></span>
<span><span class="co">##  [71] dichromat_2.0-0.1          biomaRt_2.61.3            </span></span>
<span><span class="co">##  [73] RSQLite_2.3.7              utf8_1.2.4                </span></span>
<span><span class="co">##  [75] tidyr_1.3.1                generics_0.1.3            </span></span>
<span><span class="co">##  [77] data.table_1.16.2          prettyunits_1.2.0         </span></span>
<span><span class="co">##  [79] InteractionSet_1.33.0      httr_1.4.7                </span></span>
<span><span class="co">##  [81] htmlwidgets_1.6.4          S4Arrays_1.5.11           </span></span>
<span><span class="co">##  [83] pkgconfig_2.0.3            gtable_0.3.5              </span></span>
<span><span class="co">##  [85] blob_1.2.4                 htmltools_0.5.8.1         </span></span>
<span><span class="co">##  [87] bookdown_0.41              ProtGenerics_1.37.1       </span></span>
<span><span class="co">##  [89] scales_1.3.0               png_0.1-8                 </span></span>
<span><span class="co">##  [91] knitr_1.48                 lambda.r_1.2.4            </span></span>
<span><span class="co">##  [93] rstudioapi_0.17.0          rjson_0.2.23              </span></span>
<span><span class="co">##  [95] checkmate_2.3.2            curl_5.2.3                </span></span>
<span><span class="co">##  [97] cachem_1.1.0               stringr_1.5.1             </span></span>
<span><span class="co">##  [99] parallel_4.4.1             foreign_0.8-87            </span></span>
<span><span class="co">## [101] AnnotationDbi_1.67.0       restfulr_0.0.15           </span></span>
<span><span class="co">## [103] desc_1.4.3                 pillar_1.9.0              </span></span>
<span><span class="co">## [105] grid_4.4.1                 vctrs_0.6.5               </span></span>
<span><span class="co">## [107] dbplyr_2.5.0               cluster_2.1.6             </span></span>
<span><span class="co">## [109] htmlTable_2.4.3            evaluate_1.0.1            </span></span>
<span><span class="co">## [111] VennDiagram_1.7.3          GenomicFeatures_1.57.1    </span></span>
<span><span class="co">## [113] cli_3.6.3                  locfit_1.5-9.10           </span></span>
<span><span class="co">## [115] compiler_4.4.1             futile.options_1.0.1      </span></span>
<span><span class="co">## [117] rlang_1.1.4                crayon_1.5.3              </span></span>
<span><span class="co">## [119] labeling_0.4.3             interp_1.1-6              </span></span>
<span><span class="co">## [121] forcats_1.0.0              fs_1.6.4                  </span></span>
<span><span class="co">## [123] stringi_1.8.4              deldir_2.0-4              </span></span>
<span><span class="co">## [125] munsell_0.5.1              lazyeval_0.2.2            </span></span>
<span><span class="co">## [127] csaw_1.39.0                Matrix_1.7-1              </span></span>
<span><span class="co">## [129] hms_1.1.3                  patchwork_1.3.0           </span></span>
<span><span class="co">## [131] bit64_4.5.2                KEGGREST_1.45.1           </span></span>
<span><span class="co">## [133] statmod_1.5.0              highr_0.11                </span></span>
<span><span class="co">## [135] igraph_2.1.1               broom_1.0.7               </span></span>
<span><span class="co">## [137] memoise_2.0.1              bslib_0.8.0               </span></span>
<span><span class="co">## [139] bit_4.5.0                  GenomicInteractions_1.39.0</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Kaminow2022-dz" class="csl-entry">
Kaminow, Benjamin, Sara Ballouz, Jesse Gillis, and Alexander Dobin.
2022. <span>“Pan-Human Consensus Genome Significantly Improves the
Accuracy of <span class="nocase">RNA-seq</span> Analyses.”</span>
<em>Genome Res.</em> 32 (4): 738–49.
</div>
<div id="ref-Srivastava2020-tm" class="csl-entry">
Srivastava, Avi, Laraib Malik, Hirak Sarkar, Mohsen Zakeri, Fatemeh
Almodaresi, Charlotte Soneson, Michael I Love, Carl Kingsford, and Rob
Patro. 2020. <span>“Alignment and Mapping Methodology Influence
Transcript Abundance Estimation.”</span> <em>Genome Biol.</em> 21 (1):
239.
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/smped" class="external-link">Stevie Pederson</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

    </footer>
</div>





  </body>
</html>
